### DISCUSSION TITLE
Is GPU passthrough possible on macOS VMs?

### DISCUSSION THREAD

**Discussion by samwyndham** [cirruslabs/tart#344] [Q&A] [answered]
Hi,

I would like to know about `tart`s support for GPU passthrough on macOS VMs. By GPU passthrough I mean that which is discussed on this webpage:

https://www.macstadium.com/blog/gpu-passthrough-virtualization-with-mac-private-clouds

Specifically I would like to know:

1. Does tart currently support GPU passthrough. If so how would one set it up?
2. If not, would a future version of tart likely support it?

The reason I would like to setup GPU passtrough is to test code that utilizes Apples metal APIs (via Xcode) .

Thank you!

**Comment by fkorotkov** [comment, marked as answer]
Hey Sam, 

Tart already supports paravirtualized GPUs thanks to 'Virtualization.Framework'. You can use Metal APIs inside the VMs with no additional setup. 

I'd recommend to use Ventura guest for your experiments since Apple folks drastically improved GPU support in the latest release comparing to Monterey.

**Reply by samwyndham** [reply]
Thanks @fkorotkov good to know. I will play around with a Ventura guest as you suggest.
